snomed/com.b2international.snowowl.snomed.reasoner.server/src/com/b2international/snowowl/snomed/reasoner/server/diff/SourceConceptNamespaceAndModuleAssigner.java

```diff
@@ -15,92 +15,77 @@
  * to match the source concept's namespace.
- *
  */
-public class SourceConceptNamespaceAndModuleAssigner implements NamespaceAndMolduleAssigner {
-
-	private static final Logger LOGGER = LoggerFactory.getLogger(SourceConceptNamespaceAndModuleAssigner.class);
-
-	private Map<String, Concept> conceptIdToRelationshipModuleMap = Maps.newHashMap();
-	
-	private Map<String, Concept> conceptIdToConcreteDomainModuleMap = Maps.newHashMap();
-
-	private Concept relationshipDefaultModuleConcept;
+public class SourceConceptNamespaceAndModuleAssigner implements NamespaceAndModuleAssigner {
 
-	private Concept cdDefaultModuleConcept;
+	private Map<String, Iterator<String>> namespaceToRelationshipIdMap = newHashMap();
+	private Map<String, Concept> conceptIdToRelationshipModuleMap = newHashMap();
+	private Map<String, Concept> conceptIdToConcreteDomainModuleMap = newHashMap();
 
 	@Override
-	public String getRelationshipNamespace(String sourceConceptId, final IBranchPath branchPath) {
-		return SnomedIdentifiers.create(sourceConceptId).getNamespace();
+	public String getRelationshipId(String sourceConceptId) {
+		String sourceConceptNamespace = SnomedIdentifiers.create(sourceConceptId).getNamespace();
+		return namespaceToRelationshipIdMap.get(sourceConceptNamespace).next();
 	}
 
 	@Override
-	public Concept getRelationshipModule(String sourceConceptId, final IBranchPath branchPath) {
-
-		if (!conceptIdToRelationshipModuleMap.containsKey(sourceConceptId)) {
-			LOGGER.warn("Could not find the conceptId '{}' in the allocated module map.", sourceConceptId);
-			return relationshipDefaultModuleConcept;
-		}
+	public Concept getRelationshipModule(String sourceConceptId) {
 		return conceptIdToRelationshipModuleMap.get(sourceConceptId);
 	}
 
 	@Override
-	public Concept getConcreteDomainModule(String sourceConceptId, IBranchPath branchPath) {
-		
-		if (!conceptIdToConcreteDomainModuleMap.containsKey(sourceConceptId)) {
-			LOGGER.warn("Could not find the conceptId '{}' in the allocated module map.", sourceConceptId);
-			return cdDefaultModuleConcept;
-		}
+	public Concept getConcreteDomainModule(String sourceConceptId) {
 		return conceptIdToConcreteDomainModuleMap.get(sourceConceptId);
 	}
 
 	@Override
-	public void allocateRelationshipNamespacesAndModules(Set<String> conceptIds, final SnomedEditingContext editingContext) {
-
-		//default module is the fall back
-		relationshipDefaultModuleConcept = editingContext.getDefaultModuleConcept();
-
+	public void allocateRelationshipIdsAndModules(Multiset<String> conceptIds, final SnomedEditingContext editingContext) {
+		Multiset<String> reservedIdsByNamespace = HashMultiset.create();
+		for (Multiset.Entry<String> conceptIdWithCount : conceptIds.entrySet()) {
+			String namespace = SnomedIdentifiers.create(conceptIdWithCount.getElement()).getNamespace();
+			reservedIdsByNamespace.add(namespace, conceptIdWithCount.getCount());
```

Comment:
`INT` namespaces are registered in the `Multiset` with a `null` key. It doesn't seem to hurt (the underlying `HashMap` allows using `null` as a key), but in other parts of the code we use the String literal `"INT"` for increased readability. Should I change it here as well?